edweiss-app/hooks/routeParameters.ts

```diff
@@ -0,0 +1,48 @@
+import { Href, router, useLocalSearchParams } from 'expo-router';
+import { useMemo } from 'react';
+
+/**
+ * The application route type signature alongside its path.
+ * 
+ * Use with {@link useRouteParameters} and {@link pushWithParameters}.
+ */
+export interface ApplicationRouteSignature<T> {
+	path: Href<string>
+}
+
+/**
+ * Parses the route parameters with {@link ApplicationRouteSignature} to deduce the type of the parameters.
+ * 
+ * Use {@link pushWithParameters} to push the right arguments.
+ * 
+ * @returns The parsed parameters.
+ */
+export function useRouteParameters<T>(_: ApplicationRouteSignature<T>): T {
```

Comment:
Does this allow us to push whatever object we want? Is there a limit to stringifying?

Reply:
Well you can push anything that you would have been able to with the standard `router.push`. Meaning every JSON compatible value (strings, numbers, lists, nested objects) are allowed, but functions aren't.